### DISCUSSION TITLE
What are the benefits to integrating with react-query?

### DISCUSSION THREAD

**Discussion by jestrickler** [TanStack/router#262] [Q&A]
I have been using react-query a bit and love it. I just started reading about react-location and think I will love it also. 

I know there's an example that shows them integrated, but I wonder the benefit of doing so. .?

React-query seems to fit into and really clean-up the standard render then get data workflow, where react-location cleans things up by allowing you to get the data before rendering the component. So, to me, they seem to follow different paradigms and I wonder why or if I would want to use both?

**Comment by bzbetty** [comment]
I have the same question.

From what I've seen using react query in the loader you then lose access to hooks (as loaders aren't components) and you don't get automatic rerenders when the data is invalidated.  They just don't feel like they quite play nicely together.  Which is a pity as I really like the concept of using them together and got quite excited after seeing this little project (https://github.com/oedotme/generouted)

That said i do wonder if react location is going to rethink how things work now that we have suspense.
